web/src/engine/websites/HunterScan.ts

```diff
@@ -1,21 +1,82 @@
 import { Tags } from '../Tags';
 import icon from './HunterScan.webp';
-import { DecoratableMangaScraper } from '../providers/MangaPlugin';
-import * as Madara from './decorators/WordPressMadara';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
 import * as Common from './decorators/Common';
+import { FetchJSON } from '../platform/FetchProvider';
+
+type APIResult<T> = {
+    data: T
+}
+
+type APIManga = {
+    id: string,
+    title: string,
+}
+
+type APIChapter = {
+    id: string,
+    number: string,
+    pages?: {
+        page_url: string
+    }[]
+}
 
-@Madara.MangaCSS(/^{origin}\/series\/[^/]+\/$/, 'span.rate-title')
-@Madara.MangasMultiPageAJAX()
-@Madara.ChaptersSinglePageAJAXv2()
-@Madara.PagesSinglePageCSS()
 @Common.ImageAjax()
-export default class extends DecoratableMangaScraper {
 
+export default class extends DecoratableMangaScraper {
+    private readonly apiUrl = 'https://admin.huntersscan.xyz/api/';
     public constructor() {
         super('hunterscan', 'Hunters Scan', 'https://huntersscan.xyz', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Portuguese, Tags.Accessibility.RegionLocked);
     }
 
     public override get Icon() {
         return icon;
     }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/manga/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const slug = url.match(/\/manga\/([^/]+)/)[1];
+        const request = new Request(new URL(`manga/${slug}`, this.apiUrl).href);
+        const { data } = await FetchJSON<APIResult<APIManga>>(request);
+        return new Manga(this, provider, data.id, data.title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
+    }
+
+    async getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const request = new Request(new URL(`manga?page[number]=${page}`, this.apiUrl).href);
```

Comment:
Type definitions have been updated, `Request` now supports `URL` types as well as `string`, so no longer needed to use `.href`